x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/TransportDeleteInferenceEndpointAction.java

```diff
@@ -126,9 +139,41 @@ && endpointIsReferencedInPipelines(state, request.getInferenceEndpointId(), list
         })
             .addListener(
                 masterListener.delegateFailure(
-                    (l3, didDeleteModel) -> masterListener.onResponse(new DeleteInferenceEndpointAction.Response(didDeleteModel, Set.of()))
+                    (l3, didDeleteModel) -> masterListener.onResponse(
+                        new DeleteInferenceEndpointAction.Response(didDeleteModel, Set.of(), Set.of())
+                    )
+                )
+            );
+    }
+
+    private static boolean endpointIsReferenceInPipelinesOrSemanticText(
+        final ClusterState state,
+        final String inferenceEndpointId,
+        ActionListener<Boolean> listener
+    ) {
+        logger.error("Stopping inference endpoint -- check");
+        return endpointIsReferencedInPipelines(state, inferenceEndpointId, listener)
+            || endpointIsReferenceInSemanticText(state, inferenceEndpointId, listener);
+    }
+
+    private static boolean endpointIsReferenceInSemanticText(
+        final ClusterState state,
+        final String inferenceEndpointId,
+        ActionListener<Boolean> listener
+    ) {
+        if (extractSemanticTextFields(state.getMetadata(), Set.of(inferenceEndpointId)).isEmpty() == false) {
+            listener.onFailure(
+                new ElasticsearchStatusException(
+                    "Inference endpoint "
+                        + inferenceEndpointId
+                        + " is referenced by `SemanticText` field(s) and cannot be deleted. "
+                        + "Use `force` to delete it anyway, or use `dry_run` to list the `SemanticText` field(s) that reference it.",
```

Comment:
Let's work on the language here. We may want to be more generic and just say that the inference endpoint is referenced in index mappings to be forward-looking. I'm thinking of a future where the `sparse_vector` or `dense_vector` field types have an `inference_id` param, which may happen.

Reply:
++ . Also, I think we're returning the indices that contain it, and not the actual fields.